integration_tests/src/main/python/window_function_test.py

```diff
@@ -383,6 +383,47 @@ def test_window_aggs_for_rows(data_gen, batch_size):
         conf = conf)
 
 
+@ignore_order(local=True)
+@pytest.mark.parametrize('batch_size', ['1000', '1g'], ids=idfn)
+@pytest.mark.parametrize('data_gen', [
+    [('grp', RepeatSeqGen(int_gen, length=20)),  # Grouping column.
+     ('ord', StringGen(nullable=True)),          # Order-by column (STRING).
+     ('agg', IntegerGen())]                      # Aggregation column.
+], ids=idfn)
+def test_range_windows_with_string_order_by_column(data_gen, batch_size):
+    """
+    Tests that RANGE window functions can be used with STRING order-by columns.
+    """
+    assert_gpu_and_cpu_are_equal_sql(
+        lambda spark: gen_df(spark, data_gen, length=2048),
+        'window_agg_table',
+        'SELECT '
+        ' COUNT(1) OVER '
```

Comment:
Do we want to test anything beyond just count? The rank functions are especially interesting because they look at the order by columns to determine the rank and if we now support strings they have never been tested with strings before.

If they are not supported, then it would be good to have tests that show that we are falling back correctly for them.

Reply:
Good idea. I didn't foresee there would be any impact, but it's good to add that in the test.